apps/noaabuoy/noaa_buoy.star

```diff
@@ -0,0 +1,147 @@
+    else:
+        buoy1_json = json.decode(buoy1_json)
+
+    print(str(buoy1_json))
+    height = ""
+    if "error" not in buoy1_json:
+        height = float(buoy1_json["height"])
+        if (height < 2):
+            swell_color = color_small
+        elif (height < 5):
+            swell_color = color_medium
+        elif (height < 12):
+            swell_color = color_big
+        elif (height >= 13):
+            swell_color = color_huge
+
+        height = buoy1_json["height"]
+        unit_display = "f"
+        if unit_pref == "meters":
+            unit_display = "m"
+            height = float(height) / 3.281
+            height = int(height * 10)
+            height = height / 10.0
+
+        return render.Root(
+            child = render.Box(
+                render.Column(
+                    cross_align = "center",
+                    main_align = "center",
+                    children = [
+                        render.Text(
+                            content = buoy1_name,
+                            font = "tb-8",
+                            color = swell_color,
+                        ),
+                        render.Text(
+                            content = "%s%s %ss" % (height, unit_display, buoy1_json["period"]),
+                            font = "6x13",
+                            color = swell_color,
+                        ),
+                        render.Text(
+                            content = "%s°" % (buoy1_json["direction"]),
+                            color = "#FFAA00",
+                        ),
+                    ],
+                ),
+            ),
+        )
+    else:  # if we have error key, then we got no good swell data, display the error
+        return render.Root(
+            child = render.Box(
+                render.Column(
+                    cross_align = "center",
+                    main_align = "center",
+                    children = [
+                        render.Text(
+                            content = buoy1_name,
+                            font = "tb-8",
+                            color = swell_color,
+                        ),
+                        render.Text(
+                            content = buoy1_json["error"],
+                            font = "tb-8",
+                            color = swell_color,
+                        ),
+                        render.Text(
+                            content = "Error",
+                            color = "#FFAA00",
+                        ),
+                    ],
+                ),
+            ),
+        )
+
+def get_schema():
+    unit_options = [
+        schema.Option(display = "feet", value = "feet"),
+        schema.Option(display = "meters", value = "meters"),
+    ]
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "buoy_1_id",
```

Comment:
Is there any way to get a list of Buoys based on location? We have some additional schema types that may be useful. It will likely be pretty hard for a user to find the right ID.

Reply:
There is a search function on noaa's website but I haven't got around doing anything about it.  Since this app is actually about swell/wave data and not tides (maybe a name change should happen to make that clear), most users are surfers or other mariner type people who will be familiar with specific buoy ids and know how to get them.